UI changes for unverified users

bantic opened the pull request:
Unverified users can still use the UI but they see a persistent
'you are not verified, click here to resend verification email' banner.
In addition, the 'create' buttons for apps, dbs and log drains are not show,
and the '/new' pages for each of those three shows a message saying you must be
verified to create.

fixes #128

@mixonic, for you
@sandersonet please take a look at the UI updates (comments on #128) and
let me know if you'd like to see any changes.

mixonic commented on app/apps/new/template.hbs:
```diff
@@ -1,39 +1,56 @@
-              {{/each}}
-            </div>
-          {{/unless}}
+{{#if session.currentUser.verified}}
```
I wonder if we can easily refactor this into `renderTemplate` calls on the route.

bantic replied:
good call, that works well

bantic commented:
@mixonic Some churn in the test files due to me adding some new test helpers and updating old syntax when I revisited the files.

Lots of churn to the payment test:
  * added a function to mock successful payment and pulled out explicit `fillIn` and assertions for stripe/CC data from tests that were not testing that part
  * changed the user data to default to unverified
  * updated the tests that included provisioning to set the user as verified

mixonic commented on app/components/verification-code-reset/template.hbs:
```diff
@@ -0,0 +1,27 @@
+      <span class='danger'>Your email is not activated.</span>
+
+      You will not be able to deploy or create new apps until you activate your email.
+      <button class='btn btn-default btn-xs' {{action 'resendVerification'}} {{bind-attr disabled="isDisabled"}}>
```
I think you mean `{{bind-attr disabled=isDisabled}}`, no quotes?

bantic replied:
correct, thanks

mixonic commented on app/templates/-nav-header.hbs:
```diff
@@ -10,3 +10,7 @@
 </header>
+
+{{#unless session.currentUser.verified}}
+  {{verification-code-reset store=store}}
```
Maybe we should just inject store onto components?

bantic replied:
done

bantic commented:
@mixonic addressed your comments in the last two commits. When the tests goes green this is good to go.

bantic commented:
closes #51
